tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -918,6 +918,26 @@ def program_bound_args(a: cases.IField, scalar: int32, condition: bool, out: cas
     cases.verify(cartesian_case, prog_bounds, a, out, inout=out, ref=ref)
 
 
+def test_with_bound_args_order_args(cartesian_case):
+    @gtx.field_operator
+    def fieldop_args(a: cases.IField, condition: bool, scalar: int32) -> cases.IField:
+        if not condition:
+            scalar = 0
+        return a + scalar
```

Comment:
I would simplify the test case to the minimum needed to test the problem (the condition is just distraction from the core problem).

Reply:
I would disagree here because the boolean parameter is good to asses the `type_inference` passes correctly